root/static/scripts/release-editor/actions.js

```diff
@@ -57,6 +57,14 @@ const actions = {
 
   // Information tab
 
+  selectReleaseGroup(releaseGroup) {
+    $('#release-group').val(releaseGroup.gid).trigger('input');
```

Comment:
This triggers a search for the release group's MBID. Is there any way to update the autocomplete immediately, e.g. `$('#release-group').data('mb-entitylookup').setSelection()`? We don't have the database ID, so maybe not.

Reply:
Yeah, I'm not sure if that would work correctly; `setSelection` does do `var hasID = Boolean(data.id || data.gid);`, but then later it depends on `data.id` only when pushing the item to the recent entities list...

One possibility we could look into is adding support for an `advanced` search flag to the `/ws/js/<entity>` endpoints used by the autocomplete. The results from that endpoint are fetched from the database after performing the search, and then serialized to our internal JSON format (which includes the database row ID).

It seems that would just involve reading a new `advanced` parameter in [`dispatch_search`](https://github.com/metabrainz/musicbrainz-server/blob/d3adb94/lib/MusicBrainz/Server/Controller/WS/js/Role/Autocompletion.pm#L17) and then passing it through the `$self->_indexed_search` -> `$c->model('Search')->external_search` calls.

Then you could just perform your search via this endpoint rather than `/ws/2`. Not necessary to get this merged though.

Reply:
Decided to look into this, since it wasn't too much work in the end, and pushed some commits. It seems to be working well, so I'll merge once tests pass if you don't find any issues in the meantime. 🙂